apache-airflow: 2.10.5 -> 3.1.6

taranarmo opened the pull request:
https://github.com/apache/airflow/releases/tag/3.1.6
includes changes from #393351 as dependencies
based on #396544 but heavily modified
due to pinned dependencies building invokes several hundreds packages rebuilding
additional providers may be added like
```
nix-build -I nixpkgs=. -E '(let pkgs = import <nixpkgs> {}; in pkgs.apache-airflow3.pythonPackages.apache-airflow3.override { enabledProviders = [ "airbyte" ]; } )'
```

## Things done

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform:
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- Tested, as applicable:
  - [ ] [NixOS tests] in [nixos/tests].
  - [ ] [Package tests] at `passthru.tests`.
  - [ ] Tests in [lib/tests] or [pkgs/test] for functions and "core" functionality.
- [ ] Ran `nixpkgs-review` on this PR. See [nixpkgs-review usage].
- [x] Tested basic functionality of all binary files, usually in `./result/bin/`.
- Nixpkgs Release Notes
  - [ ] Package update: when the change is major or breaking.
- NixOS Release Notes
  - [ ] Module addition: when adding a new NixOS module.
  - [ ] Module update: when the change is significant.
- [x] Fits [CONTRIBUTING.md], [pkgs/README.md], [maintainers/README.md] and other READMEs.

[NixOS tests]: https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests
[Package tests]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests
[nixpkgs-review usage]: https://github.com/Mic92/nixpkgs-review#usage

[CONTRIBUTING.md]: https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md
[lib/tests]: https://github.com/NixOS/nixpkgs/blob/master/lib/tests
[maintainers/README.md]: https://github.com/NixOS/nixpkgs/blob/master/maintainers/README.md
[nixos/tests]: https://github.com/NixOS/nixpkgs/blob/master/nixos/tests
[pkgs/README.md]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md
[pkgs/test]: https://github.com/NixOS/nixpkgs/blob/master/pkgs/test

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

taranarmo committed:
python3Packages.cadwyn: init at 5.6.1

Newer versions currently don't build with fastapi 0.121

Co-authored-by: Ethan Carter Edwards <ethan@ethancedwards.com>

taranarmo committed:
python3Packages.pytest-fixture-classes: init at 1.0.3

taranarmo committed:
python3Packages.retryhttp: init at 1.3.1

taranarmo committed:
python3Packages.issubclass: init at 0.1.2

taranarmo committed:
python3Packages.svcs: init at 25.1.0

taranarmo committed:
apache-airflow: 2.10.5 -> 3.1.0

Co-authored-by: Wolfgang Walther <walther@technowledgy.de>

taranarmo committed:
python3Packages.greenback: init at 1.2.1

wolfgangwalther commented on pkgs/development/python-modules/svcs/default.nix:
No need for the empty list, also the comment seems to be out of place:


```suggestion
  # Disable failing aiohttp tests on Darwin due to permission errors when binding to localhost
  disabledTests = lib.optionals stdenv.hostPlatform.isDarwin [
```

wolfgangwalther commented on pkgs/development/python-modules/greenback/default.nix:
```diff
@@ -0,0 +1,46 @@
+  src = fetchFromGitHub {
+    owner = "oremanj";
+    repo = "greenback";
+    rev = "v${version}";
```
These are tags upstream as well, so:


```suggestion
    tag = "v${version}";
```

Avoids problems when upstream (accidentally) pushes a branch with the same name.

wolfgangwalther commented on pkgs/development/python-modules/greenback/default.nix:
```diff
@@ -0,0 +1,46 @@
+    changelog = "https://github.com/oremanj/greenback/blob/${src.rev}/CHANGES.rst";
+    license = licenses.mit;
+    maintainers = [ ];
+  };
```
We just got rid of all `meta = with lib` in a treewide, let's not introduce more of them:


```suggestion
  meta = {
    description = "Use anyio and asyncio from synchronous code";
    homepage = "https://github.com/oremanj/greenback";
    changelog = "https://github.com/oremanj/greenback/blob/v${version}/CHANGES.rst";
    license = lib.licenses.mit;
    maintainers = [ ];
  };
```

(also fixed the changelog after the above `rev` -> `tag` change)

wolfgangwalther commented on pkgs/by-name/ap/apache-airflow3/package.nix:
```diff
@@ -0,0 +1,246 @@
+        src = fetchFromGitHub {
+          owner = "spec-first";
+          repo = "connexion";
+          rev = "refs/tags/${version}";
```
You're already using `tag` mostly consistently elsewhere, so let's do here as well:


```suggestion
          tag = version;
```

wolfgangwalther commented on pkgs/by-name/ap/apache-airflow3/package.nix:
```diff
@@ -0,0 +1,246 @@
+          pyyaml
+          requests
+          swagger-ui-bundle
+        ];
```
I believe these should be `dependencies` in python packaging (?):


```suggestion
        dependencies = with pySelf; [
          aiohttp
          aiohttp-jinja2
          aiohttp-swagger
          clickclick
          flask
          inflection
          jsonschema
          openapi-spec-validator
          packaging
          pyyaml
          requests
          swagger-ui-bundle
        ];
```

wolfgangwalther commented on pkgs/by-name/ap/apache-airflow3/package.nix:
```diff
@@ -0,0 +1,246 @@
+          pname = "trove_classifiers";
+          hash = "sha256-kxyphBpenJQIvCrme1DSis+FvvViGbVoYIdt0fLQJN0=";
+        };
+      });
```
On 2025.9.11.17 already on master.

wolfgangwalther commented on pkgs/by-name/ap/apache-airflow3/python-package.nix:
```diff
@@ -0,0 +1,470 @@
+      version = "0.0.0";
+      src = "${airflow-src}/airflow-core/src/airflow/api_fastapi/auth/managers/simple/ui";
+      fetcherVersion = 1;
+      hash = "sha256-XXA6Ynxvt/KnVT8EsV4z5Nxd5QejZO8iDDEcS04B57A=";
```
I get a different hash here for aarch64-darwin (did not try other platforms).

wolfgangwalther reviewed (commented):
What's the reason for creating a separate package instead of updating the existing `apache-airflow` to v3.x?

(build on darwin did not run through, yet, so there might be more errors than just the ones I reported)

taranarmo commented:
Upstream support for 2.x will end soon so we could just drop it (since it's not updated in nixpkgs anyway), though the main question whether do we have nixpkgs' users of AF. Popularity of this and the 2.x PR is near zero.

wolfgangwalther committed:
apache-airflow: refactor

Use replace-fail, tag and sort some deps and use
writableTmpDirAsHomeHook.

wolfgangwalther committed:
apache-airflow: remove left-overs

Once upon a time, these were important, but not anymore.

wolfgangwalther committed:
apache-airflow: expose enabledProviders

Not sure how these were supposed to be enabled, without exposing them on
the package interface itself.

wolfgangwalther committed:
apache-airflow: move update script to separate file

wolfgangwalther committed:
apache-airflow: move build dependenies to build-system

wolfgangwalther committed:
apache-airflow: 3.1.0 -> 3.1.5

taranarmo commented on pkgs/by-name/ap/apache-airflow/python-package.nix:
```diff
@@ -2,332 +2,386 @@
+
+  taskSdk = buildPythonPackage {
+    pname = "task-sdk";
+    inherit src version;
```
I believe since it's internal version of the sdk and it's binded to specific version of AF, it's not strictly correct to inherit AF version though tolerable at least for now

taranarmo committed:
apache-airflow: 3.1.5 -> 3.1.6

taranarmo committed:
python3Packages.cadwyn: disable for Python 3.14

taranarmo committed:
apache-airflow: improve update scripts

- add nixfmt-tree call to format generated providers.nix
- fix update-providers script to include correct imports
